### ISSUE TITLE
Cilium IPv6 and DualStack K8s support

### ISSUE THREAD

**Issue by fristonio** [cilium/cilium#13891]
This a parent issue that tracks the progress of Cilium on IPv6 and DualStack clusters.

Known issues:

- [x] Support for multiple `ipFamilies` in service spec. 

With Kube-Proxy

- [ ] Hairpin flow for IPv6 service (#13420)
- [x] Masquerading for IPv6 traffic (#6320) 
NodePort services with kube-proxy when service is accessed from client pod in Node 1 to Node2 NodePort address. This happens because of the asymmetric traffic routing from Node2 via the tunnel as traffic is not masqueraded. 

Kube-Proxy free mode

- [x] Validate host reachable services
- [x] Validate secondary Interface NodePort services in DualStack mode

IPv6 CI

- [x] Introduce DualStack K8s to CI and test both IPv4 and IPv6 for the same set of services.

**Comment by iam-TJ** [comment]
I think there's a related issue here; native IPv6-only with NAT64. Currently there is support for the inverse, NAT46.
We're operating IPv6 only clusters but need to support IPv4 ingress/egress.

**Comment by brb** [comment]
@iam-TJ Please create a separate feature request issue. Thanks.

**Comment by joestringer** [comment]
@fristonio @aanm Are the set of bullet points in this issue accurate? Are there any other items we would like to address for v1.10 release?

**Comment by fristonio** [comment]
Yeah, that's my understanding at this point. There is one more issue that requires the dual-stack to be enabled by default in the dev environment but I don't think it is necessarily a release blocker.

**Comment by dcorbe** [comment]
> * support for bpf-based ipv6 masquerading ?! :)
>   Currently I have to fall back to iptables based one because I'm using dual stack

Please don't try and NAT in IPv6.

**Comment by julianwiedmann** [comment]
The only remaining item tracked here ("Hairpin flow for IPv6 service") is already tracked at https://github.com/cilium/cilium/issues/26733, let's close here.
